lab_1_keywords_tfidf/main.py

```diff
@@ -99,7 +148,22 @@ def calculate_tfidf(term_freq: dict[str, float], idf: dict[str, float]) -> Optio
 
     In case of corrupt input arguments, None is returned
     """
-    pass
+    if not isinstance(term_freq, dict) or not isinstance(idf, dict) or term_freq == {}:
+        return None
+    for i in term_freq.items():
+        if not isinstance(i[0], str) or not isinstance(i[1], float):
+            return None
+    for i in idf.items():
+        if not isinstance(i[0], str) or not isinstance(i[1], float):
+            return None
+    dictionary = {}
+    for key in term_freq:
+        if key in idf:
+            tfidf = term_freq[key] * idf[key]
+        else:
+            tfidf = term_freq[key] * math.log(47 / (0 + 1))
```

Comment:
Why `(0 + 1)` ?

Reply:
thought i should write this in detail, fixed